cpp/src/gandiva/precompiled/time_test.cc

```diff
@@ -315,6 +316,18 @@ TEST(TestTime, TimeStampAdd) {
   EXPECT_EQ(add_date64_int64(StringToTimestamp("2000-02-27 00:00:00"), 4),
             StringToTimestamp("2000-03-02 00:00:00"));
 
+  EXPECT_EQ(
+      add_year_month_interval_timestamp(StringToTimestamp("2000-02-27 00:00:00"), 4),
+      StringToTimestamp("2000-06-27 00:00:00"));
+
+  EXPECT_EQ(add_day_time_interval_timestamp(StringToTimestamp("2000-02-27 00:00:00"), 4),
```

Comment:
I think other good test case to add is when the result timestamp are negative values(timestamps before `1970-01-01`)